clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -18835,6 +18835,22 @@ case Builtin::BI__builtin_hlsl_elementwise_isinf: {
     Intrinsic::ID ID = CGM.getHLSLRuntime().getWaveIsFirstLaneIntrinsic();
     return EmitRuntimeCall(Intrinsic::getDeclaration(&CGM.getModule(), ID));
   }
+  case Builtin::BI__builtin_hlsl_wave_read_lane_at: {
+    // Due to the use of variadic arguments we must explicitly retreive them and
+    // create our function type.
+    Value *OpExpr = EmitScalarExpr(E->getArg(0));
+    Value *OpIndex = EmitScalarExpr(E->getArg(1));
+    llvm::FunctionType *FT = llvm::FunctionType::get(
+        OpExpr->getType(), ArrayRef{OpExpr->getType(), OpIndex->getType()},
+        false);
+
+    // Get overloaded name
+    std::string name =
+        Intrinsic::getName(CGM.getHLSLRuntime().getWaveReadLaneAtIntrinsic(),
+                           ArrayRef{OpExpr->getType()}, &CGM.getModule());
+    return EmitRuntimeCall(CGM.CreateRuntimeFunction(FT, name, {}, false, true),
+                           ArrayRef{OpExpr, OpIndex}, "hlsl.wave.read.lane.at");
```

Comment:
The name we used for `BI__builtin_hlsl_wave_get_lane_index` was  `__hlsl_wave_get_lane_index`.  Why would we use periods here?

Reply:
I think `__hlsl_wave_get_lane_index` is the odd one out. The other intrinsics follow the pattern of `hlsl.name`. Having changed to using one word `waveReadLaneAt` I think we can keep it consistent naming with `hlsl.waveReadLaneAt`.

I can change the name to `hlsl.waveGetLaneIndex` in the clean-up pr.